### ISSUE TITLE
A chained search with reference search parameter of Any type specified with the resource type and an urn code system identifier returns an error.

### ISSUE THREAD

**Issue by j-fbriere** [jamesagnew/hapi-fhir#1419]
**Describe the bug**
A chained search with reference search parameter of Any type specified with the resource type and an urn code system identifier returns an error.
Here is the chained search: {Resource}?{searchParameter}:{ResourceType}.identifier={urnCodeSystem}|{value}

**To Reproduce**
1- Create a Resource with an identifier with an urn code system.
   Eg. 
   ```
   {
     "resourceType": "MedicationRequest",
     ...
     "identifier": [
          {
            "use": "official",
            "system": "urn:oid:2.16.124.10.101.1.60.3.30.4",
            "value": "A00000023"
          }
        ]
     ...
   }
   ```
2- Create a Task in which the focus attribute (which is a reference of Any type) specifies the resource previously created.
  Eg. 
   ```
  {
    "resourceType": "Task",
    ...
    "focus": {
          "type": "MedicationRequest",
          "identifier": {
            "system": "urn:oid:2.16.124.10.101.1.60.3.30.4",
            "value": "A00000023"
          },
          "display": "The medication request order number to be renewed"
        }
    ...
  }
   ```
3- Execute the search: `Task?focus:{ResourceType}.identifier={urnCodeSystem}|{value}`
   Eg. `Task?focus:MedicationRequest.identifier=urn:oid:2.16.124.10.101.1.60.3.30.4|A00000023`
   
4- The response is an error with the following message:
   **"Unable to perform search for unqualified chain 'focus' as this SearchParameter does not declare any target types. Add a qualifier of the form 'focus:[ResourceType]' to perform this search"**

**Expected behavior**
It should return a searchset Bundle with 0 or more resources found.

**Environment (please complete the following information):**
 - JPA Server of HAPI FHIR Version 3.8.0 with R4 model
 - JPA Server of HAPI FHIR Version 4.0.0 with R4 model

**Additional context**
Note: There is **no** error if:
      - The code system is not specified. Eg. `Task?focus:MedicationRequest.identifier=A00000023`
      - The specified code system is an URL. Eg. `Task?focus:MedicationRequest.identifier=http://terminology.hl7.org/CodeSystem/medication-type|A00000023`

Note 2: Here is the fix I made :
- Version 3.8.0:
[fix_search_with_chained_Reference_predicate_with_urn_code_system_3_8_0.patch.txt](https://github.com/jamesagnew/hapi-fhir/files/3672490/fix_search_with_chained_Reference_predicate_with_urn_code_system_3_8_0.patch.txt)
- Version 4.0.0:
[fix_search_with_chained_Reference_predicate_with_urn_code_system_4_0_0.patch.txt](https://github.com/jamesagnew/hapi-fhir/files/3672492/fix_search_with_chained_Reference_predicate_with_urn_code_system_4_0_0.patch.txt)

**Comment by j-fbriere** [comment]
In the Version 4.1.0 there is the exact same fix I proposed in this ticket.
The fix can be found in the issue [1503](https://github.com/jamesagnew/hapi-fhir/pull/1503).
Also my test shows that this issue is resolved in the 4.1.0 Version.

So I'm closing this issue as fixed for 4.1.0 Version..

**Comment by jamesagnew** [comment]
Thanks for closing! :)
